apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -1405,6 +1413,234 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_literal_expressions_in_parentheses() {
+        assert_eq!(
+            selection!("__typename: ('Product')").apply_to(&json!({})),
+            (Some(json!({"__typename": "Product"})), vec![]),
+        );
+
+        assert_eq!(
+            selection!(" __typename : 'Product' ").apply_to(&json!({})),
+            (
+                Some(json!({})),
+                vec![ApplyToError::new(
+                    "Property .\"Product\" not found in object".to_string(),
+                    vec![json!("Product")],
+                    Some((14, 23)),
+                )],
+            ),
+        );
+
+        assert_eq!(
+            selection!(
+                r#"
+                one: (1)
+                two: (2)
+                negativeThree: (-  3)
+                true: (true  )
+                false: (  false)
+                null: (null)
+                string: ("string")
+                array: ( [ 1 , 2 , 3 ] )
+                object: ( { "key" : "value" } )
+                path: (nested.path)
```

Comment:
the parentheses on this line basically do nothing, right?

Reply:
Yep, just wanted to test they are not harmful when added redundantly.